hermeto/core/package_managers/oci/oci_client.py

```
@@ -0,0 +1,383 @@
+    headers_map: dict[str, dict[str, str]],
+    concurrency: int,
+) -> None:
+    """Download blobs concurrently and verify their checksums."""
+    sem = asyncio.Semaphore(concurrency)
+
+    async def _download_one(url: str, path: Path, expected_digest: str) -> None:
+        async with sem:
+            headers = headers_map.get(url, {})
+            async with session.get(url, headers=headers) as resp:
+                if resp.status == 401:
+                    raise _TokenExpiredError(f"Token expired while downloading blob {url}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)